lib/core-classes/DeviceService.ts

```diff
@@ -110,6 +121,13 @@ export class DeviceService {
 
       results.successes.concat(successes);
       results.errors.concat(errors);
+
+      for (const deviceDocument of enrichedDocuments) {
+        await global.app.trigger(
+          'device-manager:device:attach-tenant:before',
```

Comment:
```suggestion
          'device-manager:device:attach-tenant:after',
```

Reply:
That's why we should add functional tests for every feature :sweat_smile: You could create a document with an ID computed from the device ID in a collection in a pipe on this event for example, and then just check if this document exists